packages/backend/src/vscode-youi-events.ts

```diff
@@ -199,17 +180,61 @@ export class VSCodeYouiEvents implements YouiEvents {
       if (selectedWorkspace === this.messages.open_in_a_new_workspace) {
         void vscode.commands.executeCommand("vscode.openFolder", targetFolderUri);
       } else if (selectedWorkspace === this.messages.add_to_workspace) {
-        const wsFoldersQuantity = _.size(vscode.workspace.workspaceFolders);
+        const wsFoldersQuantity = size(vscode.workspace.workspaceFolders);
         vscode.workspace.updateWorkspaceFolders(wsFoldersQuantity, null, {
           uri: targetFolderUri,
         });
+        if (wsFoldersQuantity === 0) {
+          const workspaceFileUri = this.createNewWorkspaceFile(targetFolderUri.fsPath);
+          void vscode.commands.executeCommand("vscode.openFolder", workspaceFileUri);
+        }
       }
 
       return vscode.window.showInformationMessage(successInfoMessage);
     }
     return vscode.window.showErrorMessage(errorMmessage);
   }
 
+  private createNewWorkspaceFile(targetFolderPath: string): vscode.Uri {
+    const wsFilePath = this.getUniqWorkspaceFilePath();
+    // targetFolderPath should contain only slashes
+    // all backlsashes are replaced
+    const fileContent: string = `{
+      "folders": [{
+        "path": "${targetFolderPath.replace(/\\/g, "/")}" 
+      }],
+      "settings": {
+        "actions": []
+      }
+  }`;
+
+    fs.writeFileSync(wsFilePath, fileContent);
```

Comment:
may throws exception, isn't ?

Reply:
no,
1.  it overrides an existing file if it exists
2. It is always created in user/projects folder that is writable